platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCacheDegradationController.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl
+
+import org.gradle.api.Task
+import org.gradle.api.internal.ConfigurationCacheDegradationController
+import org.gradle.api.provider.Provider
+import org.gradle.execution.plan.FinalizedExecutionPlan
+import org.gradle.execution.plan.TaskNode
+import java.util.concurrent.ConcurrentHashMap
+
+
+class DefaultConfigurationCacheDegradationController : ConfigurationCacheDegradationController {
+
+    private val tasksDegradationRequests = ConcurrentHashMap<Task, List<Provider<String>>>()
+    val currentDegradationReasons = mutableMapOf<Task, List<String>>()
```

Comment:
💭 

This should be made internal, and a method exposed to query degradation reasons, for better encapsulation